bugbug/models/bug_type_classification.py

```diff
@@ -0,0 +1,191 @@
+
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup, labels, utils
+from bugbug.model import BugModel
+
+
+# param mode if mode==False, it uses general label names
+#            elif mode==True, it uses the label names with the subcategories
+class BugTypeClassificationModel(BugModel):
+    def __init__(self, lemmatization=False, mode=False, single_class=None):
+        BugModel.__init__(self, lemmatization)
+
+        # These are all the  categories in our taxonomy
+        # both the general and
+        # the sub-categories
+        self.label_names = [
+            'API issue',
+            'Network Usage issue',
+            'Database-related issue',
+            'GUI-related issue',
+            'Performance issue',
+            'Permission-Deprecation issue',
+            'Security issue',
+            'Program Anomaly issue',
+            'Development issue'
+        ]
+
+        label_names_sub_classes = [
+            'Add-on or plug-in incompatibility',
+            'Web incompatibility',
+            'Incompatibility',
+            'Crash',
+            'Hang',
+            'Wrong Functionality',
+            'Incorrect Rendering',
+            'Test code',
+            'Compile'
+        ]
+
+
+
+        # Set the mode  ( General VS Subcategory )
+        self.mode = mode
+
+        # Set the class mode ( Single class VS All the category )
+        self.single_class = single_class
+
+        # Add the sub-classes to the classification list
+        if self.mode == True:
+            self.label_names = self.label_names + label_names_sub_classes
+
+        # Check if the element is based over the single class
+        if self.single_class is not None:
+            assert self.single_class is not None, f"unexpected class\n\tUse the args --single_class with one of the following {self.label_names}"
+            assert self.single_class in self.label_names, f"unexpected class {self.single_class}\n\tUse the args --single_class with one of the following {self.label_names}"
+            self.single_class_name = self.single_class
+            self.label_names = [self.single_class_name, "Other"]
+
+
+        self.target_size = len(self.label_names)
+
+        self.list_labels = self.label_names
+
+        if self.single_class is None:
+            self.name_file_list_labels = "bug_types"
+        else:
+            self.name_file_list_labels = "bug_test/" + self.single_class_name
+
+
+        self.calculate_importance = False
+
+        feature_extractors = [
+            bug_features.has_str(),
+            bug_features.severity(),
+            # Ignore keywords that would make the ML completely skewed
+            # (we are going to use them as 100% rules in the evaluation phase).
+            bug_features.keywords(set(self.label_names)),
```

Comment:
We need to exclude the keywords from https://github.com/mozilla/bugbug/blob/72659c4e38255b2b7bc31cf2525080b4e6cfbb15/bugbug/models/bugtype.py#L16-L31.
The strings in `self.label_names` don't actually exist as keywords